base/audit_events.go

```diff
@@ -72,6 +240,638 @@ var AuditEvents = events{
+		},
+		EventType: eventTypeAdmin,
+	},
+	AuditIDSyncGatewayCollectInfoStop: {
+		Name:               "sgcollect_info stop",
+		Description:        "sgcollect_info was stopped",
+		EnabledByDefault:   true,
+		FilteringPermitted: false,
+		MandatoryFields:    AuditFields{},
+		EventType:          eventTypeAdmin,
+	},
+	AuditIDSyncGatewayStats: {
+		Name:               "stats requested",
+		Description:        "stats were requested",
+		EnabledByDefault:   true,
+		FilteringPermitted: false,
+		EventType:          eventTypeAdmin,
+	},
+	AuditIDSyncGatewayProfiling: {
+		Name:        "profiling requested",
+		Description: "profiling was requested",
+		MandatoryFields: AuditFields{
+			"profile_type": "cpu, memory, etc.",
+		},
+		EnabledByDefault:   true,
+		FilteringPermitted: false,
+		EventType:          eventTypeAdmin,
+	},
+	AuditIDClusterInfoRead: {
+		Name:               "Sync Gateway cluster info read",
+		Description:        "Sync Gateway cluster info was viewed",
+		EnabledByDefault:   true,
+		FilteringPermitted: false,
+		EventType:          eventTypeAdmin,
+	},
+	AuditIDCreateDatabase: {
+		Name:        "Create database",
+		Description: "A new database was created",
+		MandatoryFields: AuditFields{
+			"db":     "database name",
+			"config": "JSON representation of db config",
+		},
+		EnabledByDefault:   true,
+		FilteringPermitted: false,
+		EventType:          eventTypeAdmin,
+	},
+	AuditIDReadDatabase: {
+		Name:        "Read database",
+		Description: "Information about this database was viewed.",
+		MandatoryFields: AuditFields{
+			"db": "database name",
+		},
+		OptionalFields:     AuditFields{},
+		EnabledByDefault:   true,
+		FilteringPermitted: false,
+		EventType:          eventTypeUser,
+	},
+	AuditIDDeleteDatabase: {
+		Name:        "Delete database",
+		Description: "A database was deleted",
+		MandatoryFields: AuditFields{
+			"db": "database name",
+		},
+		EnabledByDefault:   true,
+		FilteringPermitted: false,
+		EventType:          eventTypeAdmin,
+	},
+	AuditIDReadDatabaseConfig: {
+		Name:        "Read database config",
+		Description: "Database configuration was viewed",
+		MandatoryFields: AuditFields{
+			"db": "database name",
+		},
+		EnabledByDefault:   true,
+		FilteringPermitted: false,
+		EventType:          eventTypeAdmin,
+	},
+	AuditIDUpdateDatabaseConfig: {
+		Name:        "Update database config",
+		Description: "Database configuration was updated",
+		MandatoryFields: AuditFields{
+			"db":     "database name",
+			"config": "JSON representation of new db config",
+		},
+		EnabledByDefault:   true,
+		FilteringPermitted: false,
+		EventType:          eventTypeAdmin,
+	},
+	AuditIDDatabaseOffline: {
+		Name:        "Database offline",
+		Description: "Database was taken offline",
+		MandatoryFields: AuditFields{
+			"db": "database name",
+		},
+		EnabledByDefault:   true,
+		FilteringPermitted: false,
+		EventType:          eventTypeAdmin,
+	},
+	AuditIDDatabaseOnline: {
```

Comment:
For this event, do we want to log when asyncOnline completes? This is out of scope for this PR either way.

Reply:
Deferring to implementation.